kernel/kernel-api/src/main/java/io/delta/kernel/ScanFile.java

```diff
@@ -0,0 +1,145 @@
+     *       description: A map from partition column to value for this logical file. </li>
+     *    <li>name: {@code size}, type: {@code log}, description: size of the file.</li>
+     *    <li>name: {@code modificationTime}, type: {@code log}, description: the time this
+     *       logical file was created, as milliseconds since the epoch.</li>
+     *    <li>name: {@code dataChange}, type: {@code boolean}, description: When false the
+     *      logical file must already be present in the table or the records in the added file
+     *      must be contained in one or more remove actions in the same version</li>
+     *    <li>name: {@code deletionVector}, type: {@code string}, description: Either null
+     *      (or absent in JSON) when no DV is associated with this data file, or a struct
+     *      (described below) that contains necessary information about the DV that is part of
+     *      this logical file. For description of each member variable in `deletionVector` @see
+     *      <a href=https://github.com/delta-io/delta/blob/master/PROTOCOL.md#Deletion-Vectors>
+     *          Protocol</a><ul>
+     *       <li>name: {@code storageType}, type: {@code string}</li>
+     *       <li>name: {@code pathOrInlineDv}, type: {@code string}</li>
+     *       <li>name: {@code offset}, type: {@code log}</li>
+     *       <li>name: {@code sizeInBytes}, type: {@code log}</li>
+     *       <li>name: {@code cardinality}, type: {@code log}</li>
+     *    </ul></li>
+     *   </ul></li>
+     *  </ul></li>
+     * </ol>
+     */
+    public static final StructType SCAN_FILE_SCHEMA = new StructType()
+        .add("add", AddFile.SCHEMA)
+        // TODO: table root is temporary, until the path in `add.path` is converted to
+        // an absolute path.
+        .add("tableRoot", StringType.INSTANCE);
+
+
+    protected static final int ADD_FILE_ORDINAL = SCAN_FILE_SCHEMA.indexOf("add");
+
+    protected static final StructType ADD_FILE_SCHEMA =
+        (StructType) SCAN_FILE_SCHEMA.get("add").getDataType();
+
+    protected static final int ADD_FILE_PATH_ORDINAL = ADD_FILE_SCHEMA.indexOf("path");
+
+    protected static final int ADD_FILE_PARTITION_VALUES_ORDINAL =
+        ADD_FILE_SCHEMA.indexOf("partitionValues");
+
+    protected static final int ADD_FILE_SIZE_ORDINAL = ADD_FILE_SCHEMA.indexOf("size");
+
+    protected static final int ADD_FILE_MOD_TIME_ORDINAL =
+        ADD_FILE_SCHEMA.indexOf("modificationTime");
+
+    protected static final int ADD_FILE_DATA_CHANGE_ORDINAL = ADD_FILE_SCHEMA.indexOf("dataChange");
+
+    protected static final int ADD_FILE_DV_ORDINAL = ADD_FILE_SCHEMA.indexOf("deletionVector");
+
+    protected static final String TABLE_ROOT_COL_NAME = "tableRoot";
+    protected static final DataType TABLE_ROOT_DATA_TYPE = StringType.INSTANCE;
+    protected static final int TABLE_ROOT_ORDINAL = SCAN_FILE_SCHEMA.indexOf(TABLE_ROOT_COL_NAME);
```

Comment:
If these are all protected thoughts on them being here vs in InternalScanFile? What's the separation between the two

Reply:
Moved all of these to InternalScanFileUtils